src/python/pants/base/target.py

```diff
@@ -486,7 +486,7 @@ def create_sources_field(self, sources, sources_rel_path, address=None):
       filespec = sources.filespec
     else:
       sources = sources or []
-      assert_list(sources)
+      assert_list(sources,field_name='sources')
```

Comment:
We don't know the real name of this field. It needs to be passed in as a parameter to `create_sources_field()` 

We could just not warn here, but this is probably the most common code path where a user would run into this problem.